doc/Manuals/LoadBalancing/ig-xlb_x-road_external_load_balancer_installation_guide.md

```diff
@@ -0,0 +1,785 @@
+```
+
+Copy the `serverconf` database from the default instance to the new instance:
+
+```bash
+sudo -u postgres pg_dump -C serverconf | sudo -u postgres psql -p 5433 -f -
+```
+
+To avoid confusion, the *old* `serverconf` database on the master should be renamed, or even deleted.
+```bash
+sudo -u postgres psql -p 5432 -c "ALTER DATABASE serverconf RENAME TO serverconf_old";
+```
+
+
+### 4.5 Configuring the slave instance for replication
+
+Prerequisites:
+* A separate postgresql instance has been created.
+* TLS keys and certificates have been configured in `/etc/xroad/postgresql` as described in section
+[4.1 Setting up TLS certificates for database authentication](#41-setting-up-tls-certificates-for-database-authentication)
+
+
+Go to the postgresql data directory:
+ * RHEL: `/var/lib/pgsql/serverconf`
+ * Ubuntu: `/var/lib/postgresql/9.3/serverconf`
+
+Clear the data directory:
+
+ ```bash
+ rm -rf *
+ ```
+
+Do a base backup with `pg_basebackup`:
+```bash
+sudo -u postgres PGSSLMODE=verify-ca PGSSLROOTCERT=/etc/xroad/postgresql/ca.crt PGSSLCERT=/etc/xroad/postgresql/server.crt PGSSLKEY=/etc/xroad/postgresql/server.key pg_basebackup -h <master> -p 5433 -U <nodename> -D .
+```
+Add the following `recovery.conf` to the data directory:
+
+```
+standby_mode = 'on'
+primary_conninfo = 'host=<master> port=5433 user=<nodename> sslmode=verify-ca sslcert=/etc/xroad/postgresql/server.crt sslkey=/etc/xroad/postgresql/server.key sslrootcert=/etc/xroad/postgresql/ca.crt'
+trigger_file = '/var/lib/xroad/postgresql.trigger'
+```
+Set owner of the `recovery.conf` to `postgres:postgres`, mode `0600`
+
+Modify `postgresql.conf`:
+
+```
+ssl = on
+ssl_ca_file   = '/etc/xroad/postgresql/ca.crt'
+ssl_cert_file = '/etc/xroad/postgresql/server.crt'
+ssl_key_file  = '/etc/xroad/postgresql/server.key'
+
+listen_addresses = localhost
+
+# no need to send WAL logs
+# wal_level = minimal
+# max_wal_senders = 0
+# wal_keep_segments = 0
+
+hot_standby = on
+hot_standby_feedback = on
+```
+Notice that on RHEL, during `pg_basebackup` the `postgresql.conf` was copied from the master node so the WAL sender
+parameters should be disabled. Also check that `listen_addresses` is localhost-only.
+
+Start the database instance
+
+**RHEL:**
+```bash
+systemctl start postgresql-serverconf
+```
+**Ubuntu:**
+```bash
+/etc/init.d/postgresql start
+```
+Note that on Ubuntu, the command starts all configured database instances.
+
+## 5. Configuring data replication with rsync over SSH
+
+### 5.1 Set up SSH between slaves and the master
+On the master, set up a system user that can read `/etc/xroad`. A system user has their password disabled and can not log
+in normally.
+
+**Ubuntu:**
+
+```bash
+adduser --system --ingroup xroad xroad-slave
+```
+**RHEL:**
+
+```bash
+useradd -r -m -g xroad xroad-slave
+```
+
+
+Create an `.ssh` folder and the authorized keys file:
+```bash
+sudo mkdir -m 755 -p /home/xroad-slave/.ssh && sudo touch /home/xroad-slave/authorized_keys
```

Comment:
should be sudo mkdir -m 755 -p /home/xroad-slave/.ssh && sudo touch /home/xroad-slave/.ssh/authorized_keys